editor/code_editor.cpp

```diff
@@ -1325,6 +1329,9 @@ void CodeTextEditor::move_lines_up() {
 
 			int next_start_pos = text_editor->has_selection(c_next) ? text_editor->get_selection_from_line(c_next) : text_editor->get_caret_line(c_next);
 			int next_end_pos = text_editor->has_selection(c_next) ? text_editor->get_selection_to_line(c_next) : text_editor->get_caret_line(c_next);
+			if (text_editor->get_selection_to_column(c_next) == 0 && text_editor->has_selection(c_next)) {
```

Comment:
One potential pain point here is that there's 3 lines in a row which check for `text_editor->has_selection(c_next)`. Should they be merged into one `if` clause instead, same way as above (lines 1314-1324). The resulting code is of course a few good lines longer, but not necessarily any clearer.